src/lib/crypto/kimchi_backend/gadgets/common.ml

```diff
@@ -185,8 +184,41 @@ let field_of_hex (type f)
     (hex : string) : f =
   bignum_bigint_to_field (module Circuit) @@ bignum_bigint_of_hex hex
 
+(* List of field elements for each byte of hexadecimal input*)
+let field_bytes_of_hex (type f)
```

Comment:
It should complain because it needs a well formed hex, since we are not giving support for partial bits only full bytes. I will add tests capturing this